extension-vertx/src/main/java/io/smallrye/openapi/vertx/VertxAnnotationScanner.java

```diff
@@ -120,7 +120,7 @@ private void processRoutes(final AnnotationScannerContext context, OpenAPI openA
         // Get all Vert.x routes and convert them to OpenAPI models (and merge them into a single one)
         Collection<AnnotationInstance> routeAnnotations = context.getIndex()
                 .getAnnotations(VertxConstants.ROUTE);
-        List<ClassInfo> applications = new ArrayList<>();
+        Set<ClassInfo> applications = new HashSet<>();
         for (AnnotationInstance annotationInstance : routeAnnotations) {
             if (annotationInstance.target().kind().equals(AnnotationTarget.Kind.METHOD)) {
                 ClassInfo classInfo = annotationInstance.target().asMethod().declaringClass();
```

Comment:
The same class is added for every method with a route annotation. Later on every class in the list was processed with `processRouteClass(...)`. I replaced the List with a Set to avoid processing the same class multiple times. I'm not sure, if we have to maintain the order of the classes as returned by the index but AFAIK there's no well defined order.

Reply:
@MikeEdgar Is it okay to use a `HashSet` here or should I use a `LinkedHashSet` instead?

Reply:
I haven't looked into any side-affects, but probably being cautious and keeping the same iteration order that was used for the `List` would be good. Specifically, I agree with using the `LinkedHashSet` :)